rust-sdk/core/src/math/price.rs

```diff
@@ -0,0 +1,140 @@
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+use libm::{floor, pow, sqrt};
+
+use crate::U128;
+
+use super::{invert_tick_index, sqrt_price_to_tick_index, tick_index_to_sqrt_price};
+
+const Q64_RESOLUTION: f64 = 18446744073709551616.0;
+
+/// Convert a price into a sqrt priceX64
+/// IMPORTANT: floating point operations can reduce the precision of the result.
+/// Make sure to do these operations last and not to use the result for further calculations.
+///
+/// # Parameters
+/// * `price` - The price to convert
+/// * `decimals_a` - The number of decimals of the base token
+/// * `decimals_b` - The number of decimals of the quote token
+///
+/// # Returns
+/// * `u128` - The sqrt priceX64
+#[cfg_attr(feature = "wasm", wasm_bindgen(js_name = priceToSqrtPrice, skip_jsdoc))]
+pub fn price_to_sqrt_price(price: f64, decimals_a: u8, decimals_b: u8) -> U128 {
+    let power = pow(10f64, decimals_a as f64 - decimals_b as f64);
+
+    (floor(sqrt(price * power) * Q64_RESOLUTION) as u128).into()
```

Comment:
`price * power` should be `price / power` (I expected this bug should be detected by test cases 🤔)